packages/btcindexer/DOCUMENTATION.md

```diff
@@ -0,0 +1,191 @@
+
+## 1. Overview
+
+The `btcindexer` worker is a Cloudflare Worker responsible for monitoring the Bitcoin blockchain for nBTC deposits, processing them, and coordinating the minting of nBTC tokens on the Sui blockchain.
+
+## 2. Architecture
+
+```mermaid
+graph TD
+    subgraph "External Systems"
+        UI[bYield UI]
+        Relayer[Bitcoin Relayer]
+        Bitcoin[Bitcoin Network]
+        Sui[Sui Network]
+    end
+
+    subgraph "BTCIndexer Worker"
+        Router[Router]
+        Indexer[Indexer Logic]
+        SuiClient[Sui Client]
+        DB[D1 Database]
+        BlockStore[KV Block Storage]
+    end
+
+    UI -- "1. POST /nbtc (txHex)" --> Router
+    Relayer -- "2. PUT /bitcoin/blocks" --> Router
+    Relayer -- "Gets blocks from" --> Bitcoin
+
+    Router --> Indexer
+
+    Indexer -- "Stores/retrieves tx state" --> DB
+    Indexer -- "Stores/retrieves raw blocks" --> BlockStore
+    Indexer -- "Uses" --> SuiClient
+
+    SuiClient -- "3. Verifies blocks" --> Sui
+    SuiClient -- "4. Mints nBTC" --> Sui
+```
+
+The worker is composed of several key components:
+
+- **Main Worker (`src/index.ts`):** The entry point for all incoming requests. It handles HTTP requests, and scheduled cron jobs, and delegates tasks to the appropriate modules.
+- **Indexer (`src/btcindexer.ts`):** The core logic for processing Bitcoin blocks and transactions. It identifies nBTC deposits, tracks their confirmation status, and manages the minting process.
+- **Router (`src/router.ts`):** Defines the API endpoints and routes incoming requests to the correct handlers.
+- **Sui Client (`src/sui_client.ts`):** Interacts with the Sui blockchain to mint nBTC tokens.
+- **Storage (`src/storage.ts`, `src/cf-storage.ts`):** Manages data persistence using Cloudflare D1 and KV stores.
+
+## 3. State Machine
+
+The `btcindexer` tracks the state of nBTC minting transactions as they progress through the system. The state is stored in the `nbtc_minting` table in the D1 database.
+
+The `status` field of a transaction can have one of the following values:
+
+- `broadcasting`: The deposit transaction has been broadcast to the Bitcoin network, but has not yet been included in a block.
+- `confirming`: The deposit tx has been found in a Bitcoin block but does not yet have enough confirmations.
+- `finalized`: The tx has reached the required confirmation depth and is ready to be minted.
+- `minting`: The minting process on the Sui blockchain has been initiated.
+- `minted`: The nBTC has been successfully minted on the SUI network.
+- `reorg`: A blockchain reorg was detected while the tx was in the 'confirming' state. The tx block is no longer part of the canonical chain.
+- `finalized-reorg`: An edge-case status indicating that a tx was marked 'finalized', but was later discovered to be on an orphaned (re-org deeper than the confirmation depth).
+- `finalized-failed`: An attempt to mint a finalized tx failed, but it may be retried.
+
+## 4. API Endpoints
```

Comment:
let's don't do api documentation now - I've done it in https://github.com/gonative-cc/workers/pull/133